### PR TITLE
feat: Reject & Delete CSR

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This change allows users to reject or delete CSR's by utilizing the two buttons within the action menu. There is a modal popup for each button that asks the user to confirm the action.

Screenshot from 2024-06-21 16-27-01

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ui/src/app/queries.ts [hunk 1]
```diff
@@ -20,4 +20,24 @@ export async function postCSR(csr: string) {
...
+
+export async function deleteCSR(id: string) {
+    const response = await fetch("/api/v1/certificate_requests/" + id, {
+        method: 'delete',
+    })
+    if (!response.ok) {
+        throw new Error('Network response was not ok')
+    }
+    return response.json()
+}
```
Comment (gruyaume) [inline ui/src/app/queries.ts:33]: Is this expected to work? The line remains in the table after clicking on the delete button in the modal

File: ui/src/app/certificate_requests/row.test.tsx
Comment (gruyaume) [inline ui/src/app/certificate_requests/row.test.tsx:1]: It's not quite clear to me (and therefore probably to potential users) what is the difference between Deleting a Certificate Request and Rejecting a Certificate Request. We may want to clearly state what each mean in their respective confirmation modal.

File: ui/src/app/certificate_requests/row.tsx [hunk 2]
```diff
@@ -10,12 +12,42 @@ type rowProps = {
...
+        onSuccess: () => queryClient.invalidateQueries('csrs')
+    })
+
+    const mutationFunc = (mutation: UseMutationResult<any, unknown, string, unknown>) => {
+        mutation.mutate(id.toString())
+    }
+    const handleReject = () => {
+        setConfirmationModalData({
+            func: () => () => mutationFunc(rejectMutation),
+            warningText: "Are you sure you want to reject this CSR? This action cannot be undone and will remove the certificate if there is any."
```
Comment (gruyaume) [inline ui/src/app/certificate_requests/row.tsx:41]:
```suggestion
            warningText: "Are you sure you want to reject this Certificate Request? This action cannot be undone and will remove the certificate if there is any."
```

File: ui/src/app/certificate_requests/row.tsx [hunk 3]
```diff
@@ -10,12 +12,42 @@ type rowProps = {
...
+        })
+    }
+    const handleDelete = () => {
+        setConfirmationModalData({
+            func: () => () => mutationFunc(deleteMutation),
+            warningText: "Are you sure you want to delete this CSR? This action cannot be undone and will delete the certificate if there is any."
```
Comment (gruyaume) [inline ui/src/app/certificate_requests/row.tsx:47]:
```suggestion
            warningText: "Are you sure you want to delete this Certificate Request? This action cannot be undone and will delete the certificate if there is any."
```

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: ui/src/app/queries.ts (hunk 1 above)
Comment (kayra1) [reply ui/src/app/queries.ts:33]: yes. fixed.

File: ui/src/app/certificate_requests/row.test.tsx
Comment (kayra1) [reply ui/src/app/certificate_requests/row.test.tsx:1]: updated the warning texts to be a bit more descriptive

File: ui/src/app/certificate_requests/row.tsx (hunk 3 above)
Comment (kayra1) [reply ui/src/app/certificate_requests/row.tsx:47]: changed the texts a bit more

File: ui/src/app/certificate_requests/row.tsx (hunk 2 above)
Comment (kayra1) [reply ui/src/app/certificate_requests/row.tsx:41]: changed the texts to be different

**Comment by kayra1** [comment]
also the reject button is disabled now for csrs that were already rejected

**Review by saltiyazan — APPROVED** [review]

**Review by gruyaume — APPROVED** [review]

**Pull request merged**